book/src/volume-cloning.md

```diff
@@ -0,0 +1,56 @@
+# Volume Cloning
+
+## Status and Releases
+
+** Status:** Alpha
+
+Compatible with CSI Version | Supported CSI external Provisioner Version | Min k8s Version | Max k8s version
+--|--|--|--|--|--
+v1.0  | v1.3 | v1.15 | -
+
+## Overview
+
+A Clone is defined as a duplicate of an existing Kubernetes Volume.  For more information on cloning in Kubernetes see the concepts doc for [Volume Cloning](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#volume-cloning).
+
+A storage provider that allows volume cloning as a create feature, may choose to implement volume cloning via a
+control-plane CSI RPC call.
+
+## Implementing Volume cloning functionality
+
+To implement volume cloning the CSI driver MUST:
+
+1. Implement checks for `csi.CreateVolumeRequest.VolumeContentSource` in the plugins `CreateVolume` function implementation.
+2. Implement `CLONE_VOLUME` controller capability.
+
+The cloning feature in Kubernetes is implemented by adding a `DataSource` entry of `kind: PersistentVolumeClaim` to the `csi.CreateVolumeRequest`.  This presents the source volume (PersistentVolumeClaim to clone from) as a generic data source.  The source volume must:
```

Comment:
Is this information already captured in the external-provisioner and/or Kubernetes user docs?

This is more from the user's perspective, while this section is about how a vendor implements cloning in their driver.

Reply:
It is in the k8s docs and the k8s blog (or will be when 1.15 releases and things update).  I'll remove it from here, the two sets of docs are sort of colliding at this point.